drivers/dma/dma_intel_adsp_hda.c

```diff
@@ -250,6 +250,25 @@ int intel_adsp_hda_dma_start(const struct device *dev, uint32_t channel)
 
 	__ASSERT(channel < cfg->dma_channels, "Channel does not exist");
 
+#if CONFIG_PM_DEVICE_RUNTIME
+	bool first_use = false;
+	enum pm_device_state state;
+
+	/* If the device is used for the first time, we need to let the power domain know that
+	 * we want to use it.
+	 */
+	if (pm_device_state_get(dev, &state) != -ENOSYS) {
+		first_use = state != PM_DEVICE_STATE_ACTIVE;
+		if (first_use) {
```

Comment:
I don't understand why you are checking for `PM_DEVICE_STATE_ACTIVE` as `first use`.
As far as I know `pm_device_runtime_get` will not increase reference count if the operation is not successful.
So, in theory you can check for device state in the `intel_adsp_hda_dma_pm_action` and if transition should not be done return negative value. Then ref count will not be increased right:

Reply:
_**Action**_ is not called if `usage > 1`
https://github.com/zephyrproject-rtos/zephyr/blob/main/subsys/pm/device_runtime.c#L183C1-L187